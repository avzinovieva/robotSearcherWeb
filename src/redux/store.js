import appReducer from "./app-reducer";

const { createStore, combineReducers } = require("redux");

let reducers = combineReducers({
    app: appReducer
});
let store = createStore(reducers);

window.store = store;

export default store;