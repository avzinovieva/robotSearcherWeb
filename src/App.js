import React from 'react';
import './App.css';
import { setInitializeApp } from './redux/app-reducer';
import { connect } from 'react-redux';
import OrderList from './components/OrderList/OrderList';

class App extends React.Component {

  componentDidMount() {
    this.props.setInitializeApp(true)
  }

  render () {
    return (
      <div className='app-wrapper'>
        <OrderList />
        <div className='app-wrapper-content'> 
        </div>
      </div>
    );
  }
}

let mapStateToProps = (state) => {
  return {
    initialized: state.app.initialized
  }
}

export default connect(mapStateToProps, {setInitializeApp})(App);
