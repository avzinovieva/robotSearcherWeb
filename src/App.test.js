import React from 'react';
import {create} from 'react-test-renderer';
import logo from './logo.svg';
import App from './App';
import store from './redux/store';
import { Provider } from 'react-redux';

test('App matches his snapshot', () => {
  // 1. initial data
  const component = create(<Provider store={store}>
    <App />
  </Provider>);
  const tree = component.toJSON();
  // 2. expectation
  expect(tree).toMatchSnapshot();
});
