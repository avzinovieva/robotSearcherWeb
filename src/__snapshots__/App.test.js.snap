// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches his snapshot 1`] = `
<div
  className="App"
>
  HEldlo World 1
</div>
`;
